src/storage/btree.c

```diff
@@ -33301,6 +33475,153 @@ btree_online_index_dispatcher (THREAD_ENTRY * thread_p, BTID * btid, DB_VALUE *
+ *                                       online index loading.
+ *
+ * return         : Error code.
+ * thread_p (in)   : Thread entry.
+ * btid_int (in)   : B-tree info.
+ * key (int)       : Key info
+ * leaf_page (in)  : Pointer to the leaf page.
+ * search_key (in) : Search helper
+ * restart (in/out): Restart
+ * args (in/out)   : BTREE_INSERT_HELPER *.
+ */
+int
+btree_key_online_index_IB_insert_list (THREAD_ENTRY * thread_p, BTID_INT * btid_int, DB_VALUE * key,
+				       PAGE_PTR * leaf_page, BTREE_SEARCH_KEY_HELPER * search_key, bool * restart,
+				       void *other_args)
+{
+  BTREE_HELPER *helper = (BTREE_HELPER *) other_args;
+  btree_insert_list *insert_list = helper->insert_helper.insert_list;
+  DB_VALUE *curr_key;
+  int error_code = NO_ERROR;
+  bool first_insert = true;
+
+  curr_key = key;
+
+  assert (insert_list->m_key_type == btid_int->key_type);
+
+  PERF_UTIME_TRACKER time_insert_same_leaf = PERF_UTIME_TRACKER_INITIALIZER;
+  PERF_UTIME_TRACKER_START (thread_p, &time_insert_same_leaf);
+
+  while (1)
+    {
+      error_code = btree_key_online_index_IB_insert (thread_p, btid_int, curr_key, leaf_page, search_key, restart,
+						     other_args);
+      if (error_code != NO_ERROR)
+	{
+	  break;
+	}
+
+      if (helper->insert_helper.purpose != BTREE_OP_ONLINE_INDEX_IB_INSERT)
+	{
+	  assert (insert_list->m_keys_oids.size () == 1);
+	  break;
+	}
+
+      perfmon_inc_stat (thread_p, PSTAT_BT_ONLINE_NUM_INSERTS);
+      if (!first_insert)
+	{
+	  perfmon_inc_stat (thread_p, PSTAT_BT_ONLINE_NUM_INSERTS_SAME_PAGE_HOLD);
+	}
+
+      if (insert_list->next_key () != btree_insert_list::KEY_AVAILABLE)
+	{
+	  /* no more keys in list */
+	  perfmon_inc_stat (thread_p, PSTAT_BT_ONLINE_NUM_REJECT_NO_MORE_KEYS);
+	  break;
+	}
+
+      /* prepare next pair (key, oid) */
+      COPY_OID (BTREE_INSERT_OID (&helper->insert_helper), insert_list->get_oid ());
+      curr_key = insert_list->get_key ();
+
+      int key_len = btree_get_disk_size_of_key (curr_key);
+      /* assuming the key does not exist in page (an existing key requires less space,
+       * we may miss adding one more record; this is a less expensive check, we accept the 'loss' */
+      bool key_alread_in_page = false;
+      int new_ent_size = btree_get_max_new_data_size (thread_p, btid_int, *leaf_page, BTREE_LEAF_NODE, key_len,
+						      &helper->insert_helper, key_alread_in_page);
+      if (new_ent_size > spage_get_free_space_without_saving (thread_p, *leaf_page, NULL))
+	{
+	  /* no more space in page */
+	  perfmon_inc_stat (thread_p, PSTAT_BT_ONLINE_NUM_REJECT_NO_SPACE);
+	  break;
+	}
+
+      /* compare with boundary keys : NULL keys means INF bound, no check is required */
+      if (!insert_list->m_boundaries.m_is_inf_left_key)
+	{
+	  int c;
+	  c = btree_compare_key (&insert_list->m_boundaries.m_left_key, curr_key, btid_int->key_type, 1, 1, NULL);
+	  if (c != DB_LT && c != DB_EQ)
```

Comment:
shouldn't be `c != DB_GT`?

Reply:
Code is compare (left_key , curr_key);
the expected order is: left_key  <= curr_key < right_key